src/transformStateWithClones.js

```diff
@@ -8,6 +8,48 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const results = [];
+  const stateClone = {};
+
+  Object.assign(stateClone, state);
+
+  for (const action of actions) {
+    const { type } = action;
+
+    switch (type) {
+      case 'addProperties':
+        {
+          const { extraData } = action;
+
+          Object.assign(stateClone, extraData);
+
+          results.push({ ...stateClone });
+        }
+        break;
+
+      case 'removeProperties':
+        {
+          const { keysToRemove } = action;
+
+          for (const key of keysToRemove) {
+            delete stateClone[key];
+          }
+
+          results.push({ ...stateClone });
```

Comment:
As mentioned previously, pushing the state copy to the results array should be done outside of the `switch` block to adhere to the DRY principle. [DONT REPEAT YOURSELF]